lib/media_player.py

```diff
@@ -50,8 +55,14 @@ def __get_options(self):
         return options
 
     def set_asset(self, uri, duration):
-        # @TODO: HDMI or 3.5mm jack audio output
         self.player.set_mrl(uri)
+        settings.load()
+
+        # @TODO: Refactor this conditional statement.
+        if settings['audio_output'] == 'local':
```

Comment:
@vpetersson Thanks for the prompt feedback! 🙌🏽 Yeah, at first glance, it seems that I've duplicated the settings check (as seen in the `__init__` and `set_asset` methods). Without the settings check during `__init__`, the first video asset in the playlist will still use HDMI for the audio output, even though the settings say it should be via the audio jack.

I'll keep you posted. :radio:

Reply:
@vpetersson, I've just pushed the refactor recently. I tried using VLC methods like `audio_output_device_set` inside `__init__`, but didn't work. The best I could do for now is to do a refactor.

Reply:
Let's spend a few more cycles on it as it is an important feature that we can't lose.